src/matching/Matching_visitor.ml

```diff
@@ -143,6 +144,17 @@ class ['self] matching_visitor =
             (PartialDef (ent, ClassDef partial_def))
       | _ -> ()
 
+    method v_expr_as_return_stmt env e =
+      (* An implicit return is an expression whose value is being returned
+       * from a function but without the return keyword. This is not explicit
+       * in Generic so we cannot match statement patterns against these
+       * expressions. This equivalence enables that.
+       *)
+      if env.implicit_return && e.is_implicit_return then
```

Comment:
Also, is this needed once we have `m_implicit_return` in `Generic_vs_generic`? Would be good to clarify what cases do we match in each place and why are both needed.

Reply:
Yes, it's still needed. When we unconditionally add the equivalence here, we will be matching `return` vs `return` in `Generic_vs_generic`, which means `m_implicit_return` won't be called at all and this check would be bypassed in `Generic_vs_generic` if we don't have it here.

And true, it needs some clarification why. I've added more comments.